src/main/java/com/coreos/jetcd/EtcdLeaseImpl.java

```diff
@@ -0,0 +1,300 @@
+package com.coreos.jetcd;
+
+import com.coreos.jetcd.api.*;
+import com.coreos.jetcd.lease.Lease;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.grpc.ManagedChannel;
+import io.grpc.stub.StreamObserver;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of lease client
+ */
+public class EtcdLeaseImpl implements EtcdLease {
+
+    private final static int DEFAULT_TTL = 5000;
+
+    private Object lock = new Object();;
```

Comment:
Any idea why use a separate object rather than synchronizing on this? did I miss anything here?